src/settings.cpp

```diff
@@ -316,20 +320,45 @@ CriticalSection g_csConfigValues;
 } // anonymous namespace
 
 
-void Settings::DoWriteConfigValue(const char *name, const wchar_t *value)
+int __cdecl Settings::DefaultConfigRead(void *, const char *name, wchar_t* buf, size_t len)
 {
-    CriticalSectionLocker lock(g_csConfigValues);
+    //RegistryRead requires array size in bytes
+    return RegistryRead(name, buf, len * sizeof(wchar_t));
+}
 
+int __cdecl Settings::DefaultConfigWrite(void *, const char *name, const wchar_t* value)
+{
     RegistryWrite(name, value);
+    return 1; //On failure, instead of returning 0, Exceptions will be thrown from RegistryWrite
+}
+
+int __cdecl Settings::DefaultConfigDelete(void *, const char *name)
+{
+    RegistryDelete(name);
+    return 1; //On failure, instead of returning 0, Exceptions will be thrown from RegistryDelete
+}
+
+void Settings::DoWriteConfigValue(const char *name, const wchar_t *value)
+{
+    CriticalSectionLocker lock(g_csConfigValues);
+    
+    if (0 == ms_currentConfigWrite(ms_customConfigData, name, value))
+    {
+        //Only custom write function will return 0, 
+        //default function would already have thrown an exception.
+        throw std::runtime_error("User provided config write function failed");
```

Comment:
I don't see value in throwing like this — let the user-provided function do its own error handling. It's more graceful when WinSparkle silently ignores write failures, as it does now, then when it throws errors (and possibly crashes or spams with errors, the host application).

There's similarly little value in the return values if they're not actionable (although it might make sense to have them as a sort of future-proofing). It's useful in the *Read* function, which needs to fail gracefully in absence of config data, but the other are mostly ignorable.